api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/PatientTranslatorImpl.java

```diff
@@ -148,7 +148,15 @@ public org.openmrs.Patient toOpenmrsType(@Nonnull org.openmrs.Patient currentPat
 		}
 		
 		for (HumanName name : patient.getName()) {
-			currentPatient.addName(nameTranslator.toOpenmrsType(name));
+			PersonName existingName;
+			if (name.hasId()) {
+				existingName = currentPatient.getNames().stream().filter(n -> n.getUuid().equals(name.getId())).findFirst()
+				        .orElse(null);
+			} else {
+				existingName = currentPatient.getPersonName();
```

Comment:
This behaviour still strikes me as incorrectly assuming that a patient has only a single name...

Reply:
I'm on board with matching existing names based on UUIDs, but I'm not sold that this is a good idea.